src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -591,7 +591,33 @@ private function getAdjacentRecordID($offset)
 
         $map = $list->limit($limit, $limitOffset)->column('ID');
         $index = array_search($this->record->ID, $map ?? []);
-        return isset($map[$index+$offset]) ? $map[$index+$offset] : false;
+        $position = $index+$offset;
+        $page = $currentPage;
+        $lastPosition = count($map) - 1;
+        $hasPages = $this->getNumPages($gridField) > $currentPage;
+
+        if ($position === 0) {
+            $page = $currentPage > 1 ? $currentPage - 1 : $currentPage;
```

Comment:
```suggestion
        if ($position === 0 && $currentPage > 1) {
            $page =  $currentPage - 1;
```
This is a little clearer I think.

Reply:
DONE